### PR TITLE
fix: [IOBP-43] Removed now deprecated privative cards code

### PR DESCRIPTION
**Pull request by forrest57**
Depends on https://github.com/pagopa/io-app/pull/4611

## Short description
removal of privative cards from the app, although some tests still use them as a mock

## List of changes proposed in this pull request
- removal of privative cards' i18n keys
- removal of said feature's folder
- refactor of shared code whenever necessary

## How to test
`yarn jest`, also navigate to the wallet section and execute a full cobadge flow (io-dev-server is your friend here) to make sure nothing breaks.
the app should not be impacted aside from the fact that privative or "loyalty" cards have ceased to exist.

### CODE REVIEW

File: ts/components/wallet/payment/PickNotAvailablePaymentMethodListItem.tsx [hunk 1]
```diff
@@ -138,16 +138,7 @@ const extractInfoFromPaymentMethod = (
...
-        logo: paymentMethod.icon,
-        title: paymentMethod.caption,
-        description: getPickPaymentMethodDescription(
-          paymentMethod,
-          nameSurname
-        ),
-        bottomSheetTitle: unacceptedBottomSheetTitle(),
-        bottomSheetBody: unacceptedBottomSheetBody()
-      };
+      return undefined as unknown as PaymentMethodInformation;
```
Comment (mastro993) [inline ts/components/wallet/payment/PickNotAvailablePaymentMethodListItem.tsx:141]: This is not correct: `undefined` is not an `unknown` which is not a defined `PaymentMethodInformation`. We should remove `Privative` kind from `PaymentMethod` instead.

File: ts/features/bonus/bpd/saga/orchestration/__tests__/activateBpdOnNewPaymentMethods.test.tsx [hunk 2]
```diff
@@ -1,194 +0,0 @@
-import * as E from "fp-ts/lib/Either";
```
Comment (mastro993) [inline ts/features/bonus/bpd/saga/orchestration/__tests__/activateBpdOnNewPaymentMethods.test.tsx:1]: Why did you remove these tests?

File: ts/utils/paymentMethod.ts [hunk 3]
```diff
@@ -60,33 +57,24 @@ export const getPaymentMethodHash = (
       return pm.info.uidHash;
     case "Bancomat":
     case "CreditCard":
-    case "Privative":
       return pm.info.hashPan;
+    case "Privative":
+      return undefined;
```
Comment (mastro993) [inline ts/utils/paymentMethod.ts:62]: This method was not returning undefined so it should no return undefined.
Instead, remove `Privative` from the `RawPaymentMethod` kind

File: ts/utils/paymentMethod.ts [hunk 4]
```diff
@@ -271,12 +245,13 @@ export const isPaymentMethodExpired = (
...
       return E.right(false);
     case "Bancomat":
-    case "Privative":
     case "CreditCard":
       return isExpired(
         paymentMethod.info.expireMonth,
         paymentMethod.info.expireYear
       );
+    case "Privative":
+      return E.left(new Error("privative methods are deprecated"));
```
Comment (mastro993) [inline ts/utils/paymentMethod.ts:254]: Remove `Privative` from `PaymentInstrument` kind.

File: ts/screens/wallet/payment/__tests__/PickPaymentMethodScreen.test.tsx [hunk 5]
```diff
@@ -1,262 +0,0 @@
-import { AmountInEuroCents, RptId } from "@pagopa/io-pagopa-commons/lib/pagopa";
```
Comment (mastro993) [inline ts/screens/wallet/payment/__tests__/PickPaymentMethodScreen.test.tsx:1]: Why did you remove these tests?

File: ts/features/wallet/onboarding/cobadge/saga/networking/searchUserCobadge.tsx [hunk 6]
```diff
@@ -41,9 +40,9 @@ export const searchUserCobadge = async (
   sessionManager: SessionManager<PaymentManagerToken>,
   searchRequestId: string | undefined
 ): Promise<E.Either<NetworkError, CobadgeResponse>> => {
-  const logPrefix = `${
-    cobadgeQuery.panCode ? privativePrefix : cobadgePrefix
-  }_${searchRequestId ? withTokenSuffix : withoutTokenSuffix}`;
+  const logPrefix = `${cobadgeQuery.panCode ? undefined : cobadgePrefix}_${
+    searchRequestId ? withTokenSuffix : withoutTokenSuffix
+  }`;
```
Comment (mastro993) [inline ts/features/wallet/onboarding/cobadge/saga/networking/searchUserCobadge.tsx:45]:
```suggestion
  const logPrefix = `${cobadgePrefix}_${
    searchRequestId ? withTokenSuffix : withoutTokenSuffix
  }`;
```

File: ts/features/wallet/component/card/WalletV2PreviewCards.tsx [hunk 7]
```diff
@@ -49,7 +48,7 @@ const paymentMethodPreview = (
       }
       return <BPayWalletPreview key={pm.idWallet} bPay={pm} />;
     case "Privative":
-      return <PrivativeWalletPreview key={pm.idWallet} privative={pm} />;
+      return null;
```
Comment (mastro993) [inline ts/features/wallet/component/card/WalletV2PreviewCards.tsx:51]: Remove `Privative` from `PaymentMethod` kind instead

**Review by mastro993 — CHANGES_REQUESTED** [review]

File: ts/features/wallet/component/__test__/PagoPaPaymentCapability.test.tsx [hunk 8]
```diff
@@ -124,7 +124,7 @@ describe("PagoPaPaymentCapability", () => {
     );
     expect(component.getByText("Incompatible")).not.toBeNull();
   });
-  it('should render a badge with test "Incompatible" if passed a privative card, payment method of kind CreditCard with issuerAbiCode and type = PRV', () => {
+  it(' **DEPRECATED** should render a badge with test "Incompatible" if passed a privative card, payment method of kind CreditCard with issuerAbiCode and type = PRV', () => {
```
Comment (mastro993) [inline ts/features/wallet/component/__test__/PagoPaPaymentCapability.test.tsx:127]: Why? If this test is not necessary we should remove it. What do you think?

File: ts/features/wallet/onboarding/cobadge/saga/networking/searchUserCobadge.tsx [hunk 9]
```diff
@@ -18,15 +18,14 @@ type CoBadgeRequestQuery = {
...
 const withTokenSuffix = "WITH_TOKEN";
 const withoutTokenSuffix = "WITHOUT_TOKEN";
 
 /**
- * This method contains the generic networking logic used to request a user's cobadge and privative
- * cards. The request for privative cards also requires the pancode
- * @param cobadgeQuery  the input data to search the privative / cobadge
+ * This method contains the generic networking logic used to request a user's cobadge
+ * { and privative cards. The request for privative cards also requires the pancode ** DEPRECATED **}
+ * @param cobadgeQuery  the input data to search the {privative} / cobadge
```
Comment (mastro993) [inline ts/features/wallet/onboarding/cobadge/saga/networking/searchUserCobadge.tsx:28]:
```suggestion
 * This method contains the generic networking logic used to request a user's cobadge
 * @param cobadgeQuery  the input data to search the cobadge
```

**Review by mastro993 — CHANGES_REQUESTED** [review]

File: ts/features/wallet/component/__test__/PagoPaPaymentCapability.test.tsx (hunk 8 above)
Comment (forrest57) [reply ts/features/wallet/component/__test__/PagoPaPaymentCapability.test.tsx]: adressed in https://github.com/pagopa/io-app/pull/4615/commits/b9230048e2ad74c789fb95c5efbdf376e86a5186

File: ts/features/wallet/onboarding/cobadge/saga/networking/searchUserCobadge.tsx (hunk 9 above)
Comment (forrest57) [reply ts/features/wallet/onboarding/cobadge/saga/networking/searchUserCobadge.tsx]: adressed in https://github.com/pagopa/io-app/commit/b9230048e2ad74c789fb95c5efbdf376e86a5186

File: ts/features/bonus/bpd/saga/orchestration/__tests__/activateBpdOnNewPaymentMethods.test.tsx [hunk 10]
```diff
@@ -30,6 +29,8 @@ const enrollAfterAddFalse: BpdConfig = {
   opt_in_payment_methods_v2: false
 };
 
+const navFunc = jest.fn();
+
```
Comment (mastro993) [inline ts/features/bonus/bpd/saga/orchestration/__tests__/activateBpdOnNewPaymentMethods.test.tsx:33]: What's the purpose of this function? Could it be removed?
